Add utility to produce .a files

steven-johnson opened the pull request:
Work for #879: add a way to combine .o files into .a files to simplify
downstream consumption of multi-architecture output that can’t easily
be linked directly by LLVM at this time.

steven-johnson commented:
(Note that I haven't built/tested on Windows as I don't yet have a build system set up for it; relying on the buildbot to complain for now, which it probably will)

steven-johnson commented:
Expanded the tests, added deterministic mode (which is default), one more minor fix. Amazingly the Travis checks pass which I assume means we are good on Windows. PTAL.

steven-johnson commented:
(Oh, also strip paths to just basenames per spec)

abadams commented:
Travis doesn't actually test on Windows. That lot falls to the build bots.
I should really figure out how to get them to test pull requests.

On Fri, May 13, 2016 at 10:12 AM, Steven Johnson <notifications@github.com>
wrote:

> (Oh, also strip paths to just basenames per spec)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/halide/Halide/pull/1206#issuecomment-219103459>
>

steven-johnson commented:
> Travis doesn't actually test on Windows

Doh. Well, I need to get this Windows machine on my desk set up anyway...

abadams commented on src/StaticLibrary.cpp:
```diff
@@ -0,0 +1,193 @@
+};
+
+std::string pad_right(std::string s, size_t max) {
+    internal_assert(s.size() <= max) << s.size() << " "<<s;
```
nit: spaces around last <<

abadams commented on src/StaticLibrary.cpp:
```diff
@@ -0,0 +1,193 @@
+
+namespace {
+
+class File {
```
Probably belongs in Util.h or similar so people don't reinvent it next time someone wants to do cross-platform file access.

steven-johnson replied:
Agreed. Although Zalman pointed out offline that std::ofstream might be a better choice for some of this.

steven-johnson commented:
Removed File and used fstream instead; moved the static helper wrappers to Util.h as suggested. (If this lands then use of those wrappers will get propagated to existing sites.)
